### PR TITLE
Adding properties to FlywayProperties

### LINKED ISSUE #2667: Better document flyway properties
Hi there,

I cannot configure flyway the I want because the `FlywayProperties` is missing on a lot of options. Specifically I'm currently creating the flyway bean like the following:
````
    @DependsOn("dataSource")
    @Bean(name = { "defaultPlatformFlywayConfiguration", "flyway" })
    protected Flyway defaultPlatformFlywayConfiguration() throws PropertyVetoException {
        final Flyway flyway = new Flyway();
        flyway.setDataSource(dataSource);
        flyway.setInitDescription("First Platform Init");
        flyway.setTable("platform_schema_version");
        flyway.setSqlMigrationPrefix(environment.getProperty("my.platform.schema.export.prefix", "EXPORT"));
        flyway.setIgnoreFailedFutureMigration(true);
        flyway.setInitOnMigrate(true);
        flyway.setValidateOnMigrate(false);
        flyway.setLocations("classpath:/db/migration");
        flyway.setSqlMigrationSuffix(".sql");

        return flyway;
    }
````
Could you please allow customizing the `validateOnMigrate`, `initOnMigrate`, `ignoreFailedFutureMigration`, `table` and `initDescription`. 

Also the documentation suggests I can add  `flyway.sql-migration-suffix` and `flyway.sql-migration-prefix` as properties:

http://docs.spring.io/spring-boot/docs/current/reference/html/common-application-properties.html

but the file does not suggest so:

https://github.com/spring-projects/spring-boot/blob/master/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/flyway/FlywayProperties.java

Thanks.

### PR DESCRIPTION
**Pull request by eddumelendez**
Fix #2667

### CODE REVIEW

**Comment by philwebb** [comment]
Thanks! Fast turn around :)

File: spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/flyway/FlywayProperties.java [hunk 1]
```diff
@@ -72,6 +73,41 @@
...
+	 */
+	private String sqlMigrationSuffix = ".sql";
+
+	/**
+	 * Enable validation when running migrate.
+	 */
+	private boolean validateOnMigrate = true;
+
+	/**
+	 * @see Flyway#baselineOnMigrate
```
Comment (snicoll) [inline spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/flyway/FlywayProperties.java]: Don't do that please, the meta-data generation process does not handle javadoc tags.

File: spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/flyway/FlywayProperties.java [hunk 2]
```diff
@@ -72,6 +73,41 @@
...
+	/**
+	 * Set filename prefix for scripts.
+	 */
+	private String sqlMigrationPrefix = "V";
+
+	/**
+	 * Set filename suffix for scripts.
+	 */
+	private String sqlMigrationSuffix = ".sql";
+
```
Comment (snicoll) [inline spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/flyway/FlywayProperties.java]: That "Set" prefix seems wrong. _Filename prefix for scripts._ looks good to me.

File: spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/flyway/FlywayAutoConfiguration.java [hunk 3]
```diff
@@ -110,6 +111,13 @@ else if (this.flywayDataSource != null) {
...
 				flyway.setDataSource(this.dataSource);
 			}
+			flyway.setSqlMigrationPrefix(properties.getSqlMigrationPrefix());
+			flyway.setSqlMigrationSuffix(properties.getSqlMigrationSuffix());
+			flyway.setValidateOnMigrate(properties.isValidateOnMigrate());
+			flyway.setBaselineOnMigrate(properties.isBaselineOnMigrate());
+			flyway.setIgnoreFailedFutureMigration(properties.isIgnoreFailedFutureMigration());
+			flyway.setTable(properties.getTable());
+			flyway.setBaselineDescription(properties.getBaselineDescription());
 			return flyway;
```
Comment (snicoll) [inline spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/flyway/FlywayAutoConfiguration.java]: Like I said in the issue `Flyway` is exposed so besides improving the documentation, this should already work.

**Comment by eddumelendez** [comment]
I am removing those new properties in `FlywayProperties` but adding test to show the capablity to use `flyway` namespace to set more properties

**Pull request closed without merging**